### PR TITLE
[Feature Store] Fix pandas engine with default targets

### PR DESCRIPTION
**Pull request by benbd86**
Default paths  for default targets didn't work with new validation.
Default parquet target changed to be partitioned so directory path will be valid.

[ML-2277](https://jira.iguazeng.com/browse/ML-2277)
[ML-2258](https://jira.iguazeng.com/browse/ML-2258)

### CODE REVIEW

File: mlrun/datastore/targets.py [hunk 1]
```diff
@@ -121,12 +123,20 @@ def validate_target_paths_for_engine(targets, engine, source):
...
+                if isinstance(source, DataSource) and source.attributes.get(
+                    "chunksize"
+                ):
                     if is_single_file:
                         raise mlrun.errors.MLRunInvalidArgumentError(
                             "pandas CSV/Parquet targets must be a directory "
                             f"for a chunked source, got path:'{target.path}'"
                         )
+                elif target.kind == TargetTypes.parquet and target.partitioned:
+                    if is_single_file:
```
Comment (gtopper) [inline mlrun/datastore/targets.py:135]:
```suggestion
                elif target.kind == TargetTypes.parquet and target.partitioned and is_single_file:
```

File: mlrun/datastore/targets.py [hunk 2]
```diff
@@ -106,7 +107,8 @@ def validate_target_paths_for_engine(targets, engine, source):
                      else can be both single file or directory
 
     pandas:
-        if suorce contains chunksize attribute - path must be a directory
+        if source contains chunksize attribute - path must be a directory
+        else if parquet - if partitioned must be a directory
```
Comment (gtopper) [inline mlrun/datastore/targets.py:111]: I don't really understand this comment.

File: mlrun/datastore/targets.py (hunk 1 above)
Comment (benbd86) [reply mlrun/datastore/targets.py:135]: in case of parquet and partitioned - if not single file then it is valid (shouldn't go to next validation)

Comment (gtopper) [reply mlrun/datastore/targets.py:135]: But the next validation is `not is_single_file` 😄. Anyway it's not really important.

File: mlrun/datastore/targets.py [hunk 3]
```diff
@@ -121,14 +123,23 @@ def validate_target_paths_for_engine(targets, engine, source):
                     f"spark CSV/Parquet targets must be directories, got path:'{target.path}'"
                 )
             elif engine == "pandas":
-                if source.attributes.get("chunksize"):
+                if isinstance(source, DataSource) and source.attributes.get(
```
Comment (theSaarco) [inline mlrun/datastore/targets.py:126]: Please add a comment explaining why this check is needed - it's not self-apparent that the `source` may be a Pandas `DataFrame` (and the code does nothing to help with that), so at least comment about it. 
Even better - note that in the docstring of the function. 
Best - change the function signature to have a type hint - `source: Union[DataSource, pd.DataFrame]` (or something like that).

**Review by theSaarco — APPROVED** [review]
Looks good functionally - please just add some docs/comment as I requested.

**Review by Hedingber — APPROVED** [review]

**Pull request merged**